### PR TITLE
[SYCL][WIP] Extend select_device to take parameters

### PR DESCRIPTION
**Pull request by bso-intel**
This PR allows users to request a specific device of their choice,
in terms of device_type, backend, and device_num.
select_device now can take 3 OPTIONAL parameters for this purpose, so keeping backward compatibility.
When these 3 parameters are specified, the user can pin down a specific device on which they want to run a kernel.
A new env var SYCL_DEVICE_TRIPLES will also affect this api, and filters out unspecified devices.
If any device that can satisfy the parameters is not found, it will fall back to the default heuristic approach. (current implementation)

Signed-off-by: Byoungro So <byoungro.so@intel.com>

### CODE REVIEW

File: sycl/test/basic_tests/select_device.cpp [hunk 1]
```diff
@@ -0,0 +1,82 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %t.out
+// UN: env SYCL_DEVICE_TRIPLE=gpu:level0 %t.out
+// UN: env SYCL_DEVICE_TRIPLE=cpu,acc %t.out
+// UN: env SYCL_DEVICE_TRIPLE=*:opencl %t.out
+// UN: env SYCL_DEVICE_TRIPLE=*:opencl,gpu:level0 %t.out
```
Comment (s-kanaev) [inline sycl/test/basic_tests/select_device.cpp:6]: What's the `UN` here?

File: sycl/test/basic_tests/select_device.cpp [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+//
+// Checks that only designated plugins are loaded when SYCL_DEVICE_TRIPLE is
+// set. Checks that all different device types can be acquired from
+// select_device()
+// UNSUPPORTED: windows
```
Comment (s-kanaev) [inline sycl/test/basic_tests/select_device.cpp:11]: Why?
Will there be a dedicated test for windows platform?

**Comment by jbrodman** [comment]
Why are we changing the SYCL APIs?  We should not do this.

**Comment by keryell** [comment]
The name `TRIPLE` sounds both vague and misleading to me. Could we have a better name?

**Comment by jbrodman** [comment]
Please stop pursuing this path.

**Comment by bader** [comment]
@bso-intel, should we close this PR?

**Comment by bso-intel** [comment]
> @bso-intel, should we close this PR?

Yes, I will have to open another PR for similar work later.

**Pull request closed without merging**